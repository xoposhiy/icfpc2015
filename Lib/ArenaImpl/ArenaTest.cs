using System.IO;
using System.Linq;
using ApprovalTests;
using Lib.Finder;
using Lib.Intelligence;
using Newtonsoft.Json;
using NUnit.Framework;

namespace Lib.ArenaImpl
{
    [TestFixture]
    public class ArenaTest
    {
        [Test]
        public void EvaluateEdgeSolver()
        {
            EvaluateSolver(EdgeSolver(), 3);
        }

        private static void EvaluateSolver(Solver solver, int count = int.MaxValue)
        {
            var arena = new Arena(Problems.LoadProblems().Take(count).ToArray());
            var res = arena.RunAllProblems(solver);
            File.WriteAllText("arena.json", JsonConvert.SerializeObject(res, Formatting.Indented));
            Approvals.Verify(res);
        }

        private static Solver EdgeSolver()
        {
            var finder = new MagicDfsFinder();
            var solver = new Solver(finder, new MephalaOracle(finder, Metrics.ShouldNotCreateSimpleHoles));
            return solver;
        }
    }
}