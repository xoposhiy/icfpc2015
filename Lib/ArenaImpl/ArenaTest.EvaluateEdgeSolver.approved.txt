MephalaOracle-MagicDfsFinder
TotalScore: 13749
ProblemId	AvgScore	ScoreBySeeds
0	5385	5385
1	2926	2926
2	5438	5691,5529,5513,5165,6515,5333,4842,5360,5201,5237